### ISSUE TITLE
[electron] Connection to a remote backend

### ISSUE THREAD

**Issue by paul-marechal** [eclipse-theia/theia#2056]
## Goal
Being able to use the Theia desktop version running in electron to connect to a remote backend.

## Approaches
1. Load an electron specific frontend from the remote backend
2. Use the local Electron frontend and redirect the websocket connections on the remote backend

While (2) was prototyped, we could foresee problems with mismatching Theia's versions: the JSON-RPC protocol transfered over the Websocket might break because of some changes made between the versions.

On the other hand (1) makes the browser dependent on Electron. But part of the work will be to maybe make this remote connection feature an extension that could be supported or not based on the needs of both the host and the client.

## To be done
- [ ] Add an extension for the "Browser version" to enable the backend to serve Electron's frontend.
- [x] Add an extension for the Electron version to add the UI elements to connect to a remote.
- [ ] Change the way the Electron version manages its windows in order to deactivate the Node integration and other options in regards to security concerns raised by the loading of remote content.

**Comment by Ten0** [comment]
What's the current state of this issue?
Is there an ETA for the feature? What would be the next steps? Is it very hard?

**Comment by marechal-p** [comment]
I did realize a prototype, but we decided to not go forward with it at the moment. The main issue is security, loading remote content inside a browser window is not recommended when using electron's api + node integration. On the other hand, the electron frontend code doesn't work really well out of the box without electron + node apis, and some monkey patching needs to be made so that some components don't try to use `require` dynamically, for example.

Closed PR: https://github.com/eclipse-theia/theia/pull/2340

TL;DR: Can be done, but it feels sketchy to me for now. A lot of care needs to be taken in regards to security when doing this.

**Comment by frank-dspeed** [comment]
- [ ] https://github.com/eclipse-theia/theia/issues/8493 is related

**Comment by JonasHelming** [comment]
We have SSH remot support now, can we close this? @paul-marechal

**Comment by msujew** [comment]
@JonasHelming Yes, I'll close this. For everyone still interested in this issue. Please see https://github.com/eclipse-theia/theia/issues/12390.

**Comment by paul-marechal** [comment]
@JonasHelming yes this can be closed thanks to the amazing work by @msujew on the SSH support :)
